Documentation: Add steps to install clang

LalitSinghRana opened the pull request:
In our build instructions, we provide command for installing everything **except a compiler**. This PR adds instruction on how to install a compatible version of **clang**.

This is purely to help newer devs (less familiar with c++ and Linux ecosystem), to make their onboarding experience better. As seen in last week in `#build-problems` channel in discord.

BertalanD commented:
Is the `clang-18` package even available on Ubuntu versions older than 24.04?

There's the [apt.llvm.org](https://apt.llvm.org) repository if people want to install newer Clang versions than shipped by their distro.

circl-lastname commented:
We should probably document apt.llvm.org as an option

LalitSinghRana commented:
> Is the clang-18 package even available on Ubuntu versions older than 24.04?

I tested on 24 and 22. So, it works on those at least.

> We should probably document apt.llvm.org as an option

If we are fine adding a `wget` to an external link, we should ignore this and add instructions for `apt.llvm.org`. I wasn't sure if we allow commands with external links or not. **I can undo this and add instruction for that**.

ADKaster commented:
Pointing developers to `apt.llvm.org` is fine with me, that's what I use :D.

Additionally, we use `ppa:ubuntu-toolchain-r/test` in CI for the latest gcc.

The steps we recommend should be the 'proper' ones though, using  the `[signed-by` attribute in the sources.list file, rather than the 'old way' with `apt-key add -`

e.g. how I did it for the devcontainer

https://github.com/LadybirdBrowser/ladybird/blob/8c09b9a2e9d25230e4a26ab7c9a27fe978b71a16/.devcontainer/features/ladybird/install.sh#L20-L24

ADKaster commented:
Also do note that we don't use conventional commits here. The commit message (not the PR title, the PR title is irrelevant to our workflow) should start with the category. So in this case: 

`Documentation: Add steps to install clang` or similar.

To update this please don't close the PR and open an new one, but instead amend your commit locally and force-push to your branch.

LalitSinghRana commented:
1. Removed old instructions and added instructions to use `apt.llvm.org`
2. Tested these new instructions on ubuntu **24.04**, **22.04** and **20.04**. Successfully installed clang-18 on each.

We can add `g++` installation instruction in different PR.

---

Also improved formatting for Debian/Ubuntu section only for now, can do other sections separately.

![image](https://github.com/LadybirdBrowser/ladybird/assets/31415645/3ccf3747-89e0-4d22-9a17-a62a2a2d4f41)

circl-lastname commented:
IMO apt.llvm.org should be mentioned as a backup if your Debian-like distro doesn't ship a new enough C compiler slready

LalitSinghRana commented:
Currently only ubuntu 24.04 ships with compatible clang-17/g++13. Debian-12 ships with clang-14/g++12.

So, using `apt.llvm.org` is necessary for all, except for devs on latest ubuntu-24.04.
